src/main/java/seedu/duke/reflection/ReflectionManager.java

```diff
@@ -0,0 +1,216 @@
+package seedu.duke.reflection;
+
+import seedu.duke.exception.BadCommandException;
+import seedu.duke.exception.InvalidCommandException;
+import seedu.duke.manager.Manager;
+
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+public class ReflectionManager extends Manager {
+    private static final String FEATURE_NAME = "Self Reflection";
+    private static final String BRIEF_DESCRIPTION = "Users can get a random set of questions to reflect on.";
+    private static final String FULL_DESCRIPTION = "";
+    private static final String GET_COMMAND = "get";
+    private static final String GET_PAYLOAD = "";
+    private static final String RETURN_MAIN = "return";
+    private static final String RETURN_PAYLOAD = "";
+    private static final String EXIT_COMMAND = "exit";
+    private static final String EXIT_PAYLOAD = "";
+    private static final String NO_ELEMENT_MESSAGE = "There is no new line of input, please key in inputs.";
+    private static final String INVALID_COMMAND_MESSAGE = "Please enter a valid command.";
+    private static final String EMPTY_COMMAND_MESSAGE = "The command is empty, please check your input.";
+    private static final ReflectUi UI = new ReflectUi();
+
+    // I need to set this as static if I want to set it to true if ExitCommand object.
+    // If I create another object in ExitCommand, the corresponding isExit will be for a new object,
+    // not the one we intend to terminate.
+    // Anyone has other ideas??
+    private static boolean isExit;
```

Comment:
Can check the way i implemented my exit function in atomic habits using an instance of exit command to check.

Reply:
I leave it like this first, if there is any spooky behaviours I'll try ur method:D